src/templateengine.vala

```diff
@@ -86,7 +86,7 @@ namespace Valdo.TemplateEngine {
             if (file_type == FileType.DIRECTORY) {
                 // Test if Directroy name is a template
                 var folder_name = (!)project_child.get_basename ();
-                if (folder_name.has_prefix ("${")) {
+                if (folder_name.contains ("${")) {
```

Comment:
I think you might want to get rid of this `if` and just unconditionally substitute the directory name.